src/NuGet.Core/NuGet.Configuration/Settings/CredentialsItem.cs

```diff
@@ -0,0 +1,331 @@
+                        _validAuthenticationTypes.Value = value;
+                    }
+                }
+            }
+        }
+
+        protected override bool CanHaveChildren => true;
+
+        internal override bool IsEmpty() => string.IsNullOrEmpty(Username) && string.IsNullOrEmpty(Password);
+
+        internal AddItem _username { get; set; }
+
+        internal AddItem _password { get; set; }
+
+        internal AddItem _validAuthenticationTypes { get; set; }
+
+        public CredentialsItem(string name, string username, string password, bool isPasswordClearText, string validAuthenticationTypes)
+           : base()
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
+            }
+
+            if (string.IsNullOrEmpty(username))
+            {
+                throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(username));
+            }
+
+            if (string.IsNullOrEmpty(password))
+            {
+                throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(password));
+            }
+
+            ElementName = name;
+
+            _username = new AddItem(ConfigurationConstants.UsernameToken, username);
+
+            var passwordKey = isPasswordClearText ? ConfigurationConstants.ClearTextPasswordToken : ConfigurationConstants.PasswordToken;
+            _password = new AddItem(passwordKey, password);
+
+            IsPasswordClearText = isPasswordClearText;
+
+            if (!string.IsNullOrEmpty(validAuthenticationTypes))
+            {
+                _validAuthenticationTypes = new AddItem(ConfigurationConstants.ValidAuthenticationTypesToken, validAuthenticationTypes);
+            }
+        }
+
+        internal CredentialsItem(XElement element, SettingsFile origin)
+            : base(element, origin)
+        {
+            ElementName = element.Name.LocalName;
+
+            var elementDescendants = element.Elements();
+            var countOfDescendants = elementDescendants.Count();
+
+            if (countOfDescendants != 2 && countOfDescendants != 3)
+            {
+                throw new NuGetConfigurationException(string.Format(CultureInfo.CurrentCulture, Resources.UserSettings_UnableToParseConfigFile, origin.ConfigFilePath));
+            }
+
+            var parsedItems = elementDescendants.Select(e => SettingFactory.Parse(e, origin) as AddItem).Where(i => i != null);
+
+            foreach (var item in parsedItems)
+            {
+                if (string.Equals(item.Key, ConfigurationConstants.UsernameToken, StringComparison.OrdinalIgnoreCase))
+                {
+                    _username = item;
+                }
+                else if (string.Equals(item.Key, ConfigurationConstants.PasswordToken, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (_password != null)
```

Comment:
We have never enforced this on code (i.e. failed reading) but we have never used more than one password if present, and it is implicit in our documentation and in the way we write the credentials that only one of this should be present. I could add validation also to make sure there is only one `_username` and one `_validAuthenticationTypes`.

What do you mean with this pattern being enforced more generally? `CredentialsItem` is making its own rules about how it should be constructed, each item can do the same. As for general add items in a section, [this](https://github.com/NuGet/NuGet.Client/pull/2370/files#diff-92bbf31cd5aa989dca530c722d6a93d1R2694) existing tests says that we should ignore duplicates instead of failing. Do you think is a better idea to follow this on `CredentialsItem` and just ignore any duplicates?